enrolment/static/javascripts/index.js

```
@@ -416,20 +423,39 @@ GOVUK.components = (new function() {
    **/
   this.CompaniesHouseNameLookup = CompaniesHouseNameLookup;
   function CompaniesHouseNameLookup($input, $field) {
+    var instance = this;
     SelectiveLookup.call(this,
       $input,
       GOVUK.data.getCompanyByName
     );
 
     // Some inner variable requirement.
     this._private.$field = $field || $input; // Allows a different form field to receive value.
+    this._private.$form = $input.parents("form");
+    this._private.$errors = $(".errors", this._private.$form);
+    
+    // Custom error handling.
+    this._private.$form.on("submit.CompaniesHouseNameLookup", function(e) {
+      // If no input or no company selected
+      if(instance._private.$field.val() === "") {
+        e.preventDefault();
+        instance._private.$errors.append("<p>Check that you entered the company name correctly and select the matching company name from the list.</p>");
```

putting HTML in javascript hast the disadvantages that it adds complexity when grepping for string (I can see a developer in  future going "oh, that html is added dynamically by index.js!" after scratching their head for a while)

to avoid that, we can move this html into the django html template, and hide is by default, and have this js instead selectively show/hide that error.

This also has the advantage that human-readable text is managed soley by django templates, meaning localisation and translations only need to happen in one place (the templates) rather than two (templates and js).